### DISCUSSION TITLE
Get data from a user's OIDC authorization?

### DISCUSSION THREAD

**Discussion by thetayloredman** [outline/outline#6665] [Self Hosting]
I have users authenticating with OIDC (Discord) and I need a webhook which has access to the user's Discord user ID. How can I use the Outline API to obtain this? (Or do I need to somehow look it up with Discord by email, which would probably require me storing the authorization codes?)

**Comment by tommoor** [comment]
Hi @thetayloredman – currently there is not any webhook or API that exposes this information, one would need to be added. If you'd like to work on that I can provide some pointers for an implementation that would be accepted

**Reply by thetayloredman** [reply]
I'm not exactly sure how to expose this without it being a security risk—especially if things like the code grant are what get exposed. Is there a field in OIDC responses that is akin to Discord's user IDs that could be used and shared without an issue?

**Reply by thetayloredman** [reply]
@tommoor I think this would be best done perhaps just exposing all of the claims from the OIDC identity endpoint into some user data field?
